manage_filter_edit_page.php

```diff
@@ -0,0 +1,207 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Manage filter edit page
+ *
+ * @package MantisBT
+ * @copyright Copyright 2016  MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ * @link http://www.mantisbt.org
+ *
+ * @uses core.php
+ * @uses access_api.php
+ * @uses authentication_api.php
+ * @uses compress_api.php
+ * @uses config_api.php
+ * @uses constant_inc.php
+ * @uses current_user_api.php
+ * @uses custom_field_api.php
+ * @uses filter_api.php
+ * @uses filter_constants_inc.php
+ * @uses gpc_api.php
+ * @uses helper_api.php
+ * @uses html_api.php
+ * @uses lang_api.php
+ * @uses print_api.php
+ * @uses string_api.php
+ * @uses version_api.php
+ */
+
+require_once( 'core.php' );
+require_api( 'access_api.php' );
+require_api( 'authentication_api.php' );
+require_api( 'compress_api.php' );
+require_api( 'config_api.php' );
+require_api( 'constant_inc.php' );
+require_api( 'current_user_api.php' );
+require_api( 'custom_field_api.php' );
+require_api( 'filter_api.php' );
+require_api( 'filter_constants_inc.php' );
+require_api( 'gpc_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'html_api.php' );
+require_api( 'lang_api.php' );
+require_api( 'print_api.php' );
+require_api( 'string_api.php' );
+require_api( 'version_api.php' );
+
+auth_ensure_user_authenticated();
+
+layout_page_header( lang_get('manage_filter_edit_page_title' ) );
+
+layout_page_begin( 'manage_filter_edit_page.php' );
+
+$f_filter_id = gpc_get_int( 'filter_id', null );
+if( null === $f_filter_id ) {
+	error_parameters( 'FILTER_ID' );
+	trigger_error( ERROR_EMPTY_FIELD, ERROR );
+}
+
+$t_filter_string = filter_db_get_filter( $f_filter_id );
+if( !$t_filter_string ) {
+	access_denied();
+} else {
+	$t_filter = filter_deserialize( $t_filter_string );
+	$t_filter['_source_query_id'] = $f_filter_id;
+	filter_cache_row( $f_filter_id );
```

Comment:
At the moment there is not a clean api to load the filter in one go, so that block of code is a bit raw, and repeated in some places.
Ideally, in the future this should be restructured into a more transparent api operation.

filter_db_get_filter( $f_filter_id ) returns the serialized string
filter_deserialize( $t_filter_string ) converts it into a filter properties array
filter_cache_row( $f_filter_id ) loads the data related to owner, etc